maas/resource_maas_instance.go

```diff
@@ -45,6 +45,13 @@ func resourceMAASInstance() *schema.Resource {
 				Description: "Nested argument with the constraints used to machine allocation. Defined below.",
 				Elem: &schema.Resource{
 					Schema: map[string]*schema.Schema{
+						"architecture": {
```

Comment:
Do we want to restrict the architectures to the set MAAS allows, or rely on the api response to error if you supply, say `arm82`

Reply:
Is there any way of getting this information ahead-of-time? This sounds like it needs to be an API call, one I'm not sure is publicly exposed or implemented. Maybe best to let the `terraform apply` fail due to the MAAS machine allocate failing as that's a guarantee, and leave a FIXME above?